simul/platform/runsimul.go

```diff
@@ -55,11 +61,19 @@ func Simulate(serverAddress, simul, monitorAddress string) error {
 		if err != nil {
 			return err
 		}
-		err = sim.Node(sc)
-		if err != nil {
-			return err
-		}
 		sims[i] = sim
+		func(sc *onet.SimulationConfig) {
+			server.RegisterProcessorFunc(simulInitID, func(env *network.Envelope) {
+				err = sim.Node(sc)
+				if err != nil {
+					log.Error(err)
+				}
+				sc.Server.Send(env.ServerIdentity, &simulInitDone{})
+			})
+			server.RegisterProcessorFunc(simulInitDoneID, func(env *network.Envelope) {
+				wgSimulInit.Done()
+			})
+		}(sc)
```

Comment:
Mhhh I see. If you simply create a variable to store the current value of sc **inside the loop**, it should work without the `func()`
```
for sc .... {
   mySc := sc
   RegisterProcessor(mySc)
}
```
The tmp variable is ony available from the current iteration so should not change. At least it does not in [go playground](https://play.golang.org/p/BwsqXYcJ81)